src/nvim/api/win_config.c

```diff
@@ -381,54 +415,80 @@ static bool parse_float_bufpos(Array bufpos, lpos_T *out)
   return true;
 }
 
-static void parse_border_title(Object title, FloatConfig *fconfig, Error *err)
+static void parse_bordertext(Object bordertext, BorderTextType bordertext_type, FloatConfig *fconfig, Error *err)
 {
-  if (title.type == kObjectTypeString) {
-    if (title.data.string.size == 0) {
-      fconfig->title = false;
+  bool *is_present;
+  VirtText *chunks;
+  int *width;
+  int default_hl_id;
+  if (bordertext_type == kBorderTextTitle) {
+    is_present = &fconfig->title;
+    chunks = &fconfig->title_chunks;
+    width = &fconfig->title_width;
+    default_hl_id = syn_check_group(S_LEN("FloatTitle"));
+  } else if (bordertext_type == kBorderTextFooter) {
+    is_present = &fconfig->footer;
+    chunks = &fconfig->footer_chunks;
+    width = &fconfig->footer_width;
+    default_hl_id = syn_check_group(S_LEN("FloatFooter"));
+  }
```

Comment:
Similar to my other comment, if a new `BorderTextType` variant is ever added you could end up with an uninitialized read from this `if` branch. Same advice as before: either convert to `switch` or initialize pointers to `NULL` and `assert` after the if statement.